### ISSUE TITLE
Page crashes when injected provider is used.

### ISSUE THREAD

**Issue by MicahZoltu** [Uniswap/uniswap-interface#1447]
**Bug Description**
Attempting to use an Inejected provider (wallet), the page crashes (white screen with error in console).

**Steps to Reproduce**

1. Navigate to https://app.uniswap.org
2. Click Connect to a Wallet button.
3. Choose Injected
4. Notice that the page turns white and you get an error like:
```
TypeError: Cannot read property 'detectNetwork' of undefined
    at r.value (json-rpc-provider.ts:331)
    at new r (base-provider.ts:487)
    at r.<anonymous> (createSuper.js:12)
    at new r (json-rpc-provider.ts:308)
    at r.<anonymous> (createSuper.js:12)
    at new r (web3-provider.ts:119)
    at ZS (getLibrary.ts:5)
    at provider.tsx:52
    at Object.Eo [as useMemo] (react-dom.production.min.js:168)
    at t.useMemo (react.production.min.js:23)
```

**Expected Behavior**
Page doesn't crash.

**Additional Context**
I suspect that something in the code is making assumptions about what the injected provider provides beyond what is standard.  In particular, it seems to want to get network information from the injected provider but is failing.

There are a few issues here:
1. Uniswap is not gracefully handling the exception and presenting it to the user.
2. Uniswap is making assumptions about what features the injected provider will have that are beyond the scope of what is standardized across wallets.
3. Uniswap is not gracefully degrading when a particular feature is missing.

**Comment by tomteman** [comment]
Tom from Portis here, please let us know if there is any way we can help. 
For example, if there is functionality we can add on our end to better support your flow, we'd be happy to implement it.

**Comment by cammellos** [comment]
Hi, 
we have the same issue on the status dapp browser.
It looks like it might be an issue with ethers.js 
https://github.com/ethers-io/ethers.js/issues/1484 

As far as I can see uniswap is serving 5.1.0, and this should be fixed on 5.1.3

**Comment by moodysalem** [comment]
we have this issue with a number of wallets, it was caused by switching to a newer version of ethers. this workaround should fix it for at least coinbase wallet
https://github.com/Uniswap/uniswap-interface/commit/d3898cf900962bab6ba33e742cc66d40e59ff9fd

**Comment by moodysalem** [comment]
crash starts on line 305 here https://unpkg.com/browse/@ethersproject/providers@5.1.2/src.ts/json-rpc-provider.ts

**Comment by cammellos** [comment]
Thanks @moodysalem 
Indeed the issue was on that line, just out of curiosity, have you deployed the changes?

I have tried on coinbase wallet & status browser and I get an empty page when accessing app.uniswap.org, but unlike before no error is thrown (though page is completely blank)

The last request made is this one:

```
request: {"method":"eth_chainId","params":[]}
```

(I have only looked at the logs on status as I can't access them on the coinbase wallet).

Thanks for the prompt response.

**Comment by moodysalem** [comment]
@ricmoo we had to workaround a bug in the latest ethers provider with this commit

**Comment by moodysalem** [comment]
should be fixed in interface

**Comment by ricmoo** [comment]
@moodysalem That was fixed in [5.1.3](https://github.com/ethers-io/ethers.js/blob/master/CHANGELOG.md#ethersv513-2021-04-19-2101). The current version is 5.1.4.

Updating to the latest version should solve this. Let me know if there are still issues after updating. :)
